tests/unit/mainwindow/test_tabwidget.py

```diff
@@ -108,7 +109,7 @@ def test_pinned_size(self, widget, fake_web_tab, config_stub,
 
         for i in range(num_tabs):
             if i in pinned_num and shrink_pinned and not vertical:
-                assert (first_size.width() <
+                assert (first_size.width() >
```

Comment:
Oh! Was this a bug in the test, but it didn't show up because the tab bar was too narrow? I think this makes more sense now indeed, as `first_size.width()` is the width of an unpinned tab, right?

Reply:
So, I'm actually kind of torn about this. Previously, I was filling the tabs enough so pinned tabs were larger than normal tabs. This wasn't affected by the change of `tabSizeHint` of over-provisioning because the size was actually never set correctly. 

Because tabSizeHint will always give the width of the tabbar now for unpinned tabs, this dosen't actually do much, but I threw it in just in case something really broken happens with the pinned tabs. The test below this one is the one that actually is doing real checking for us :P